RevenueCatUI/Templates/V2/Components/ComponentsView.swift

```diff
@@ -34,36 +43,43 @@ struct ComponentsView: View {
-            case .purchaseButton(let viewModel):
-                PurchaseButtonComponentView(viewModel: viewModel)
-            case .stickyFooter(let viewModel):
-                StickyFooterComponentView(viewModel: viewModel)
-            case .timeline(let viewModel):
-                TimelineComponentView(viewModel: viewModel)
-            case .tabs(let viewModel):
-                TabsComponentView(viewModel: viewModel, onDismiss: onDismiss)
-            case .tabControl(let viewModel):
-                TabControlComponentView(viewModel: viewModel, onDismiss: onDismiss)
-            case .tabControlButton(let viewModel):
-                TabControlButtonComponentView(viewModel: viewModel, onDismiss: onDismiss)
-            case .tabControlToggle(let viewModel):
-                TabControlToggleComponentView(viewModel: viewModel, onDismiss: onDismiss)
+        ForEach(Array(componentViewModels.enumerated()), id: \.offset) { index, item in
+            Group {
+                switch item {
+                case .root(let viewModel):
+                    RootView(viewModel: viewModel, onDismiss: onDismiss)
+                case .text(let viewModel):
+                    TextComponentView(viewModel: viewModel)
+                case .image(let viewModel):
+                    ImageComponentView(viewModel: viewModel)
+                case .icon(let viewModel):
+                    IconComponentView(viewModel: viewModel)
+                case .stack(let viewModel):
+                    StackComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .button(let viewModel):
+                    ButtonComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .package(let viewModel):
+                    PackageComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .purchaseButton(let viewModel):
+                    PurchaseButtonComponentView(viewModel: viewModel)
+                case .stickyFooter(let viewModel):
+                    StickyFooterComponentView(viewModel: viewModel)
+                case .timeline(let viewModel):
+                    TimelineComponentView(viewModel: viewModel)
+                case .tabs(let viewModel):
+                    TabsComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .tabControl(let viewModel):
+                    TabControlComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .tabControlButton(let viewModel):
+                    TabControlButtonComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                case .tabControlToggle(let viewModel):
+                    TabControlToggleComponentView(viewModel: viewModel, onDismiss: onDismiss)
+                }
+            }
+            // Applies a top padding to mimmic safe area insets
+            // This was designed to be applied to the 
+            .applyIf(index > 0 && self.ignoreSafeArea) { view in
```

Comment:
suuuuuper nit:
Whenever possible I prefer to avoid things like `applyIf` in favor of:

```swift
.padding(.top, (index > 0 && self.ignoreSafeArea) ? self.safeAreaInsets.top : 0)
```

	•	The view structure remains stable because .padding(.top, value) is always present.
	•	The padding is updated and SwiftUI handles efficiently.

This avoid unnecessary view hierarchy changes

Reply:
Ah yes! Will fix 😊 This used to be more complex in the `applyIf` but then it simplified to just one thing. Thanks for catching!